senders/prometheus.go

```diff
@@ -120,15 +124,17 @@ func (s *PrometheusSender) PrecisionTiming(metric metrics.PrecisionTimingMetric)
 	return nil
 }
 
-func (s *PrometheusSender) getOrCreateAppMetrics(guid string) appMetrics {
-	m, present := s.appMetrics[guid]
+func (s *PrometheusSender) getOrCreateAppMetrics(guid string, instance string) appMetrics {
+	guidInstance := fmt.Sprintf("%s:%s", guid, instance)
+
+	m, present := s.appMetrics[guidInstance]
```

Comment:
Shouldn't we change the name of the type of the variable `appMetrics` to `appInstanceMetrics` as they are now representing metrics per instance. This is defined happens all across `prometheus.go` 

Also, a small suggestion: you might want to create a new `type instanceMetricsStore map[string]appInstanceMetrics` and add to it the methods `Get(key)` (instead of `getOrCreateAppMetrics()`) and `Delete(Key)`.

Reply:
We've added a commit to address the first issue. With regards to the 2nd, we understand what you mean for `get` and `delete` but want to double check what you're suggesting for creating `appInstanceMetrics`, would this be a separate method, or that `get` would also be responsible for creating `appInstanceMetrics` if they don't exist already?

Reply:
>  that get would also be responsible for creating appInstanceMetrics if they don't exist already?

That, it would be the same.

But that is just a suggestion, do not worry.

Reply:
I would merge this, unless you want to change something else.